packages/babel-plugin-transform-class-properties/src/index.js

```diff
@@ -41,9 +41,9 @@ export default function ({ types: t }) {
         let nodes = [];
         let ref;
 
-        if (path.isClassExpression()) {
-          ref = path.scope.generateUidIdentifier();
-        } else { // path.isClassDeclaration()
+        if (path.isClassExpression() || !path.node.id) {
+          ref = path.scope.generateUidIdentifier("class");
+        } else { // path.isClassDeclaration() && path.node.id
           ref = path.node.id;
         }
```

Comment:
Ah your right - I guess we could do `let ref = path.node.id` and remove the else but its just a style thing - I think we can merge